runtime/oti/VMHelpers.hpp

```diff
@@ -2115,17 +2119,16 @@ class VM_VMHelpers
 	static VMINLINE bool
 	tryWinningConcurrentGCScan(J9VMContinuation *continuation, bool isGlobalGC)
 	{
-		uintptr_t complementGCConcurrentState = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
+		bool ret = false;
+		volatile uintptr_t *localAddr = &continuation->state;
+		uintptr_t complementGCConcurrentState = *localAddr;
 		uintptr_t returnedState = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
-		do {
-			/* preserve the concurrent GC state for the other type of GC */
-			complementGCConcurrentState = continuation->state & getConcurrentGCMask(!isGlobalGC);
-			returnedState = VM_AtomicSupport::lockCompareExchange(&continuation->state, complementGCConcurrentState, complementGCConcurrentState | getConcurrentGCMask(isGlobalGC));
-		/* if the other GC happened to change its concurrentGC state since us taking a snapshot of their state, we'll have to retry */
-		} while (complementGCConcurrentState != (returnedState & complementGCConcurrentState));
-
-		/* if returned state does not contain carrier ID, return that we won */
-		return (complementGCConcurrentState == returnedState);
+		while (!ret && (!isConcurrentlyScannedFromContinuationState(complementGCConcurrentState, isGlobalGC)) && (!isContinuationMountedFromContinuationState(complementGCConcurrentState))) {
+			returnedState = VM_AtomicSupport::lockCompareExchange(localAddr, complementGCConcurrentState, complementGCConcurrentState | getConcurrentGCMask(isGlobalGC));
```

Comment:
returnedState does not escape while loop any more, so it could be declared here, rather then before while loop